### PR TITLE
Allow conversations to be destroyed

### LINKED ISSUE #1324: Ability to delete PMs from your inbox
Users will want to be able to delete private messages from their inbox to avoid it becoming overly cluttered, but they can't right now.

(Is there an issue for general PM inbox styling/functionality reform? I thought there was, but I can't find it.)

### PR DESCRIPTION
**Pull request by parrish**
Closes #1324

### CODE REVIEW

File: app/talk/inbox-conversation.cjsx [hunk 1]
```diff
@@ -67,6 +67,11 @@ module?.exports = React.createClass
       .then (message) =>
         @setConversation()
 
+  handleDelete: (e) ->
+    e.preventDefault()
```
Comment (aweiksnar) [inline app/talk/inbox-conversation.cjsx]: Looks great, only request is to add a window.confirm to the handler since it's a harsh action

**Comment by parrish** [comment]
Can do.  By the way, the conversation "revives" if a user replies again, so it's not really gone until all participants delete it

**Comment by aweiksnar** [comment]
Ah perfect, or we could just change it to 'Hide' or something then

**Comment by parrish** [comment]
"Archive?"

**Comment by aweiksnar** [comment]
:+1: Yeah that works, @vrooje was on board with 'archive' here #1324

**Pull request merged**